### DISCUSSION TITLE
Is it possible to export paper to .dwg (AutoCad) file?

### DISCUSSION THREAD

**Discussion by Reshin-P** [clientIO/joint#1720] [Q&A]
Hi everyone!

I'm developing webapp using JointJS and right now I'm having difficulties in  features.
 
Is it possible to export paper to .dwg (AutoCad) file? 

How I can implement Undo and Redo option in joint js ? 

How to export to svg and download it?

If there is any options to do this .

How can I do this?  Any suggestions?

Kind regards.

**Comment by kumilingus** [comment]
> Is it possible to export paper to .dwg (AutoCad) file?

JointJS does not support exporting the paper to DWG.

You'd have to export the paper to SVG and find a library to convert SVG to DWG.
 
> How I can implement Undo and Redo option in joint js ?

Please see this answer:  #1654

> How to export to svg and download it?

The `paper.svg.outerHTML` returns a serialized SVG of the paper, turn it into a blob, and download it with [util.downloadBlob()](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#util.downloadBlob).

Note, there is a commercial JointJS+ [SVG](https://resources.jointjs.com/docs/rappid/v3.5/format.SVG.html) plugin available which also converts all images from the paper into a dataURLs and allows you to export only a part of the diagram.
